File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    int answer = 0;
    
    public int solution(int[] numbers, int target) {
        System.out.println(numbers.length);
        dfs(0, 0, numbers, target);
        
        return answer;
    }
    
    public void dfs(int index, int sum, int[] numbers, int target){
        if(index == numbers.length && sum == target) answer++;
        if(index >= numbers.length) return;
        
        dfs(index+1, sum + numbers[index], numbers, target);
        dfs(index+1 , sum - numbers[index], numbers, target);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let answer = 0;

const dfs = (numbers, target, index, sum) => {
    if(index === numbers.length){
        if(target === sum) answer++;
        return;
    }
    dfs(numbers, target, index+1, sum + numbers[index]);
    dfs(numbers, target, index+1, sum - numbers[index]);
}

function solution(numbers, target) {
    dfs(numbers, target, 0, 0);
    return answer;
}